mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        # Check if required files exist
+
+        file_dat = glob.glob(fname + '/*.dat')
+        file_evt = glob.glob(fname + '/*.evt')
+        file_hdr = glob.glob(fname + '/*.hdr')
+        file_inf = glob.glob(fname + '/*.inf')
+        file_set = glob.glob(fname + '/*.set')
+        file_tpl = glob.glob(fname + '/*.tpl')
+        file_wl1 = glob.glob(fname + '/*.wl1')
+        file_wl2 = glob.glob(fname + '/*.wl2')
+        file_cfg = glob.glob(fname + '/*config.txt')
+        file_mat = glob.glob(fname + '/*probeInfo.mat')
+
+        assert_one(file_dat, "Should be one dat file")
+        assert_one(file_evt, "Should be one evt file")
+        assert_one(file_hdr, "Should be one hdr file")
+        assert_one(file_inf, "Should be one inf file")
+        assert_one(file_set, "Should be one set file")
+        assert_one(file_tpl, "Should be one tpl file")
+        assert_one(file_wl1, "Should be one wl1 file")
+        assert_one(file_wl2, "Should be one wl2 file")
+        assert_one(file_cfg, "Should be one config file")
+        assert_one(file_mat, "Should be one mat file")
+
+        # Read number of rows of wavelength data which corresponds to
+        # number of samples
+        last_sample = -2
+        for line in open(file_wl1[0]):
+            last_sample += 1
+
+        # Read participant information file
+
+        inf = cp.ConfigParser(allow_no_value=True)
+        inf.read(file_inf)
+        inf = inf._sections['Subject Demographics']
+
+        # mne requires specific fields in here
+        # https://github.com/mne-tools/mne-python/ ...
+        # blob/master/mne/io/meas_info.py#L430
+        # TODO: Can you put more values in subject_info than specified in link?
+        #       NIRX also records "Study Type", "Experiment History",
+        #       "Additional Notes", "Contact Information"
+        subject_info = {}
+        subject_info['last_name'] = inf['name'].split()[-1].replace("\"", "")
+        subject_info['first_name'] = inf['name'].split()[0].replace("\"", "")
+        subject_info['middle_name'] = inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = 1
+        if subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = 2
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # This is a bit tricky as the header file isnt compliant with
+        # the config specifications. So we need to remove all text
+        # between comments before passing to config parser
+        hdr_str = open(file_hdr[0]).read()
```

Comment:
```suggestion
        with open(file_hdr[0]).read() as hdr_str:
```
This will auto-close when the context manager exist (incl. on error)